### ISSUE TITLE
Defining bsoncodec.StructTagParser

### ISSUE THREAD

**Issue by sashker** [Kamva/mgm#7]
Hello.

Pretty often I use included structures from other packages for which I can't specify `bson:"name` tag (without redefining all of them). As a result, the mongodb driver applies ugly names to fields and structures.
I'd like to implement my own StructTagParser as mentioned here (https://jira.mongodb.org/browse/GODRIVER-607)
```If you want to customize this behavior for everything, you can create a new bsoncodec.StructTagParser and assign it to bsoncodec.DefaultStructTagParser or use it while creating a new bsoncodec.StructTagCodec and register that codec with the *bsoncodec.Registry that you use.```

But I'm not quite sure how to apply that CustomStructTagParser when I use the mgm client. Maybe, there are easier ways to do so, but CustomStructTagParser looks pretty straightforward.

Basically, I want to rely on **json** tag when **bson** tag is not specified for a structure field.

Thanks in advance.

**Comment by mehran-prs** [comment]
Hi @sashker 

 - To set your parser globally, `bsoncodec.DefaultStructTagParser` works here also.
- To set specific registry(contains your custom parser) on all of the mgm collections with default connection, just set your registry on initializing your mgm client:  
```go
_ = mgm.SetDefaultConfig(nil, "my_db", options.Client().SetRegistry(myRegistryWithCustomParser).ApplyURI("{MongoDB URI}"))
```
- To set Your custom registry on specific collection using `CollectionGetter` interface (so all of your calls to `mgm.Coll()` will use this collection):  
```go
// Book is our Model
type Book struct {
	mgm.DefaultModel `bson:",inline"`
	Name             string `json:"name" bson:"name"`
}

func (b *Book) Collection() *mgm.Collection {
    _,_,db,_:=mgm.DefaultConfigs()
    return mgm.NewCollection(db,mgm.CollName(b),options.Collection().SetRegistry(myRegistryWithCustomParser))
}
```
- To set custom registry without `CollectionGetter` interface:  
```go
_,_,db,_:=mgm.DefaultConfigs()
mgm.NewCollection(db,collectionName,options.Collection().SetRegistry(myRegistryWithCustomParser))
```
or simpler (available in __v3__):
```
coll:= mgm.Coll(&Book{},options.Collection().SetRegistry(myRegistryWithCustomParser))
```
Yes Sashker, CustomStructTagParser is good, but I think we don't need to it because of available options provided by MongoDB driver. What do you think?

**Comment by sashker** [comment]
Hi, @mehran-prs 

Thank you for pointing me out. Yes, I agree that SetRegistry() is the most native way to specify your own Parser and other custom logic of the bsoncodec.

Did you mean that by ```CustomStructTagParser is good, but I think we don't need to it because of available options provided by MongoDB driver```? Or there is some other way to change behavior of the default client?

**Comment by mehran-prs** [comment]
you're welcome!
Yes, I mean `setRegistry`

**Comment by sashker** [comment]
Just in case, if somebody encounters the same problem, the possible solution might be

https://play.golang.org/p/qYGExjF00vx
